dbt/adapters/bigquery/python_submissions.py

```diff
@@ -121,17 +123,21 @@ def _get_batch_id(self) -> str:
     def _submit_dataproc_job(self) -> dataproc_v1.types.jobs.Job:
         batch = self._configure_batch()
         parent = f"projects/{self.credential.execution_project}/locations/{self.credential.dataproc_region}"
+        batch_id = uuid.uuid4().hex
 
-        request = dataproc_v1.CreateBatchRequest(
-            parent=parent,
-            batch=batch,
-            batch_id=self._get_batch_id(),
-        )
+        request = dataproc_v1.CreateBatchRequest(parent=parent, batch=batch, batch_id=batch_id)
         # make the request
-        operation = self.job_client.create_batch(request=request)  # type: ignore
+        self.job_client.create_batch(request=request)
         # this takes quite a while, waiting on GCP response to resolve
         # (not a google-api-core issue, more likely a dataproc serverless issue)
-        response = operation.result(polling=self.result_polling_policy)
+
+        state = "State.PENDING"
+        while state not in ["State.SUCCEEDED", "State.FAILED", "State.CANCELLED"]:
```

Comment:
Pretty sure this is an enum defined some where in the bigquery python client. We should rely on that and not hardcoded strings.